AudioContextInterruptedState/explainer.md

```diff
@@ -0,0 +1,212 @@
+- `"interrupted"` -> `"running"`;
+  - By the time that the cause of the interruption ceases to exist, the UA can transition to `"running"` if audio playback is allowed to resume automatically.
+- `"interrupted"` -> `"suspended"`; and
+  - By the time that the cause of the interruption ceases to exist, the UA can transition to `"suspended"` if audio playback is **not** allowed to resume automatically or if [`AudioContext.suspend()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-suspend) has been called during the interruption.
+- `"interrupted"` -> `"closed"`.
+  - The `AudioContext`'s state should move immediately to `"closed"` when `AudioContext.close()` is called.
+
+The state transition from `"suspended"` to `"interrupted"` should not happen automatically, due to privacy concerns. Doing this transition automatically every time an interruption occurs might unnecessarily expose too much information to web pages - e.g., when a phone call comes in. Given this, what should happen if an interruption happens while the `AudioContext` is in the `"suspended"` state? In this case, an internal boolean flag, let's say `"is_interrupted"`, should be set and the AudioContext should remain in the `"suspended"` state, even though it is interrupted behind the curtains. However, calling [`AudioContext.resume()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-resume) should trigger the state change from `"suspended"` to `"interrupted"` if the interruption is still active. Therefore, while in the `"suspended"` state with an ongoing interruption:
+- Calling `AudioContext.resume()` would return a rejected promise and transition to the `"interrupted"` state;
+- Calling `AudioContext.suspend()` is a NOOP;
+
+Finally, while in the `"interrupted"` state:
+- Calling `AudioContext.resume()` would return a rejected promise;
+- Calling `AudioContext.suspend()` would change the `AudioContext`'s state to `"suspended"` and return a promise that should resolve when the `AudioContext`'s state has transitioned to `"suspended"`.
+- Calling `AudioContext.close()` would return a promise that should resolve when the `AudioContext`'s state has transitioned to `"closed"`
+
+## Key scenarios
+
+The `"interrupted"` state can be used by a couple of proposed API's, like the [`"media-playback-while-not-visible"`](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/IframeMediaPause/iframe_media_pausing.md) permission policy and the  [Audio Session API](https://w3c.github.io/audio-session/) (this list is not exhaustive).
+
+### "media-playback-while-not-visible" permission policy
+
+Whenever an iframe which has the `"media-playback-while-not-visible"` enabled is not rendered anymore, the UA would transition the iframe's AudioContext to the interrupted state to pause audio rendering. This would allow the UA to pause audio playback being rendered by an `AudioContext`.

```

Comment:
Also missing some commas here:
"Whenever an iframe which has the `"media-playback-while-not-visible"` enabled is not rendered anymore, "
-->
"Whenever an iframe, which has the `"media-playback-while-not-visible"` enabled, is not rendered anymore,

Reply:
nit: Is this sentence redundant and unnecessary?

"This would allow the UA to pause audio playback being rendered by an `AudioContext`."